gspread/worksheet.py

```diff
@@ -361,13 +362,49 @@ def range(self, name=""):
             for j, value in enumerate(row)
         ]
 
+    def _get_size_from_range(self, range_name):
+        """
+        GSHEET API returns a list of lists but it strips empty TRAILING cells
+        and strips empty TRAILING rows
+        """
+
+        # find out the width and height of the range from `data_range`
+        start_cell, end_cell = range_name.split(':')
+        start_col, start_row = re.findall(r'\d+|\D+', start_cell)
+        end_col, end_row = re.findall(r'\d+|\D+', end_cell)
+        # the columns can be A-Z or AA-ZZ or AAA-ZZZ
+        # so we need to convert the column letters to numbers
+        # and then calculate the number of columns
+        if len(start_col) == 1:
+            # e.g ord("A") = 65, ord("B") = 66, ord("C") = 67
+            num_cols = ord(end_col) - ord(start_col) + 1
+        else:
+            # In the case of multiple letters,
+            # we use the letters at each position as "digits"
+            # in a base-26 representation
+            start_col_int = 0
+            for i, col in enumerate(start_col[::-1]):
+                start_col_int += (ord(col) - 64) * (26 ** i)
+            
+            end_col_int = 0
+            for i, col in enumerate(end_col[::-1]):
+                end_col_int += (ord(col) - 64) * (26 ** i)
+
+            num_cols = end_col_int - start_col_int + 1
+            
+        # calculate the number of rows
+        num_rows = int(end_row) - int(start_row) + 1
+
+        return num_rows, num_cols
```

Comment:
we already have a function that does that computation :wink: 

see https://github.com/burnash/gspread/blob/4b4239fcd09a5312d5e5e9c32ccbaf86d9a84c7f/gspread/utils.py#L386

Reply:
does the function `a1_range_to_grid_range` work with named ranges? (see https://github.com/burnash/gspread/issues/947#issuecomment-1688552030)

e.g., what happens when the user asks for the range `"A3:B"` (wanting an expanding column) or just `"Sheet1"`. How can we know the size of the data requested?